src/wallet/externalsigner.h

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2018 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_WALLET_EXTERNALSIGNER_H
+#define BITCOIN_WALLET_EXTERNALSIGNER_H
+
+#include <stdexcept>
+#include <string>
+#include <univalue.h>
+#include <util/system.h>
+
+struct PartiallySignedTransaction;
+
+class ExternalSignerException : public std::runtime_error {
+public:
+    using std::runtime_error::runtime_error;
+};
+
+//! Enables interaction with an external signing device or service, such as
+//! a hardware wallet. See doc/external-signer.md
+class ExternalSigner
+{
+private:
+    //! The command which handles interaction with the external signer.
+    std::string m_command;
+
+public:
+    //! @param[in] command      the command which handles interaction with the external signer
+    //! @param[in] fingerprint  master key fingerprint of the signer
+    //! @param[in] mainnet      Bitcoin mainnet or testnet
+    ExternalSigner(const std::string& command, const std::string& fingerprint, bool mainnet, std::string name);
+
+    //! Master key fingerprint of the signer
+    std::string m_fingerprint;
+
+    //! Bitcoin mainnet or testnet
+    bool m_mainnet;
```

Comment:
Is a bool enough here? E.g. what if someone wants to use external signers on signet, regtest? Or are those all "testnet" in this description?

Reply:
I should update the comment, but yes: signet, testnet and regtest use the same address format, which is the only thing that seems to matter when using hardware wallets (at the least the devices I've worked with). They don't even have their own BIP44/49/81 coin types.

Reply:
I overlooked that regtest has a different prefix. Maybe an enum is a good idea...
https://github.com/satoshilabs/slips/blob/master/slip-0173.md#registered-human-readable-parts